app/pages/sections/catalogue-solutions/order-item/controller.js

```diff
@@ -206,17 +206,76 @@ export const postSolutionOrderItem = async ({
     selectedPrice,
     formData,
   });
+
+  await postData({
+    endpoint, body, accessToken, logger,
+  });
+  logger.info(`Order item successfully created for order id: ${orderId}`);
+  return { success: true };
+};
+
+const formatPutData = ({
+  formData,
+}) => ({
+  deliveryDate: extractDate('deliveryDate', formData),
+  quantity: parseInt(formData.quantity, 10),
+  estimationPeriod: formData.selectEstimationPeriod,
+  price: parseFloat(formData.price),
+});
+
+const putSolutionOrderItem = async ({
+  accessToken,
+  orderId,
+  orderItemId,
+  formData,
+}) => {
+  const endpoint = getEndpoint({ api: 'ordapi', endpointLocator: 'putCatalogueOrderItem', options: { orderId, orderItemId } });
+  const body = formatPutData({
+    formData,
+  });
+
+  await putData({
+    endpoint, body, accessToken, logger,
+  });
+  logger.info(`Order item successfully updated for order id: ${orderId} and order item id: ${orderItemId}`);
+  return { success: true };
+};
+
+export const saveSolutionOrderItem = async ({
+  orderId,
+  orderItemId,
+  accessToken,
+  selectedRecipientId,
+  serviceRecipientName,
+  selectedSolutionId,
+  solutionName,
+  selectedPrice,
+  formData,
+}) => {
   try {
-    await postData({
-      endpoint, body, accessToken, logger,
-    });
-    logger.info(`Order item successfully created for order id: ${orderId}`);
-    return { success: true };
+    const response = orderItemId === 'newsolution'
+      ? await postSolutionOrderItem({
+        accessToken,
+        orderId,
+        selectedRecipientId,
+        serviceRecipientName,
+        selectedSolutionId,
+        solutionName,
+        selectedPrice,
+        formData,
+      })
+      : await putSolutionOrderItem({
+        accessToken,
+        orderId,
+        orderItemId,
+        formData,
+      });
+    return response;
   } catch (err) {
     if (err.response.status === 400 && err.response.data && err.response.data.errors) {
       return err.response.data;
     }
-    logger.error(`Error creating order item for order id: ${orderId}`);
+    logger.error(`Error saving order item for order id: ${orderId}`);
```

Comment:
Which order item had an error saving? Do you know at this point?

Reply:
we do know the orderItemId if it's the put. But if it's a post we don't know the orderItemId. But I'll add the solutionName and serviceRecipient that errored